Add the basic harness implementation

glesica opened the pull request:
Here's the thing. Let's talk about it.

glesica committed:
Add the basic harness implementation

taylorreiter commented on harness/data/haystack.fa:
```diff
@@ -0,0 +1 @@
+atgtgatgccatgcacttacaaagaactatccatcagagctaccgaattacgcggagttacgttcggggtcttaatagcatttgaggaaatgtctgcaga
```
```suggestion
>my_fasta_1
atgtgatgccatgcacttacaaagaactatccatcagagctaccgaattacgcggagttacgttcggggtcttaatagcatttgaggaaatgtctgcaga
```

switch format to proper fasta format

glesica replied:
The mock search doesn't understand fasta, but we definitely want this to be a valid file of one sort or another (depending on which formats we decide to support)

taylorreiter replied:
awesome, thanks!

glesica committed:
Update harness/data/haystack.fa

Co-authored-by: Taylor Reiter <taylorreiter@gmail.com>

glesica committed:
Remove fasta extension for clarity

glesica committed:
Update tool to use a wrapper

glesica committed:
Excise fasta

glesica committed:
Remove unnecessary volume declaration

glesica committed:
Use the right comment syntax

glesica committed:
Add run instructions

glesica committed:
Update readme

Commit:
Add Dockerfile, readme, and wrapper script for thecount

colligant committed:
Update README.md

colligant committed:
Update README.md

glesica commented on harness/the_count/wrapper.sh:
```diff
@@ -0,0 +1,12 @@
+  echo "usage: wrapper.sh <needle.fa> <haystack.fa>"
+  exit
+fi
+# todo: should this script build the image?
```
Nope, it's going to be run from inside the container.

glesica committed:
Spruce up wrapper

Commit:
Change the_count's image to one hosted by traviswheelerlab

glesica committed:
Use better data

glesica committed:
Add AWS config

glesica committed:
Update mock search for new design

glesica committed:
Rename data to fixtures

glesica committed:
Ignore secrets

glesica committed:
Update fixture naming conventions

glesica committed:
Add transmark script

glesica committed:
Update scripts for aws

Commit:
Add working mock-search example

glesica committed:
Resolve weird symlink issues for docker

Commit:
Add George's changes to make tarring work

Commit:
Remove unecessary code and fix a few filenames

Commit:
Remove readlines()

Commit:
Revert fromPath to value

glesica committed:
Merge pull request #4 from NCBI-Codeathons/new-python-code

Add working mock-search example

glesica committed:
Use real data with mock tool

glesica committed:
Update readme

glesica committed:
Remove old params file

Commit:
Add gscc config and todo in transmark-run

Commit:
Add sleep command and cluster configuration

Commit:
Add 4 alignments to each query file

traviswheeler committed:
Remove redundant config; try to force image pull

Cleaning things up a bit. For Tommy's previous commit, he
added a sleep command to the docker image (so we could see
it running on the cluster), and pushed it to dockerhub ...
but when I run this script, I don't get the updated image.

The nested docker config (which I've commented) is supposed
to make the 'docker run' command force a pull of an
updated version of the image, is '--pull always' is set ...
but it raises an error inside nextflow.  When I imanually
run the .command.sh command ... it works as it should.
More work is required here.

glesica commented:
I already changed it @traviswheeler, is there a specific place I missed? There are two commands, 'prepare_data' and 'run_tool'.

traviswheeler committed:
Remove unnecessary custom slurm flag

traviswheeler committed:
Add optional 'docker run --pull' config

In the case that a docker image on cluster compute nodes is
out of date, I've added a (commented) config setting that can
force an image update on all compute nodes. Only works on
docker 20.10.0 or higher

Also refactored a coulpe small naming issues.

glesica commented:
I'm not seeing that many newlines...

```
1.0	2.0	5840	6163	rmark1	TRAIN.23S_rRNA_IVP.12
1.0	2.0	8614	8937	rmark1	TRAIN.23S_rRNA_IVP.12
1.0	2.0	9661	9984	rmark1	TRAIN.23S_rRNA_IVP.12
1.0	2.0	9738	10061	rmark1	TRAIN.23S_rRNA_IVP.12



psss-team3-hard-annotation/harness on  harness-spike [!?] took 10s
➜
```

That's the bottom of the output. Looks like two newlines. Were you seeing this with Slurm? Maybe that runner introduces some extra stuff?

We can easily write those to a file, but for now it's just more convenient to see them so that we can test quickly as we iterate on it. Ultimately, we're going to be feeding that output into post-processing, so that will probably mean writing it out to a file.

glesica committed:
Revert to docker by default

glesica committed:
Remove the_count

traviswheeler committed:
Added a bit to the README and documentation
